datamanager/sqlite_data_manager.py

```diff
@@ -263,6 +262,8 @@ def update_movie(self, user_movie_id, updated_details):
         """
         try:
             user_movie = self.get_user_movie(user_movie_id)
+            if not user_movie:
+                return {"error": "This movie is not in your list."}
```

Comment:
Just to show you there is the 'walrus operator' since Python 3.8 where you can write this even more compact if you want (https://docs.python.org/3/whatsnew/3.8.html):

```suggestion
            if not user_movie := self.get_user_movie(user_movie_id):
                return {"error": "This movie is not in your list."}
```